client/app/components/ApplicationArea/ErrorMessage.jsx

```diff
@@ -1,17 +1,43 @@
-import React from "react";
+import { isObject, get } from "lodash";
+import React, { useMemo } from "react";
 import PropTypes from "prop-types";
 
-export default function ErrorMessage({ error, showOriginalMessage }) {
+function getErrorMessageByStatus(status, defaultMessage) {
+  switch (status) {
+    case 404:
+      return "It seems like the page you're looking for cannot be found.";
+    case 401:
+    case 403:
+      return "It seems like you don’t have permission to see this page.";
+    default:
+      return defaultMessage;
+  }
+}
+
+export function getErrorMessage(error) {
+  const message = "It seems like we encountered an error. Try refreshing this page or contact your administrator.";
+  if (isObject(error)) {
+    // HTTP errors
+    if (error.isAxiosError && isObject(error.response)) {
+      return getErrorMessageByStatus(error.response.status, get(error, "response.data.message", message));
+    }
+    // Router errors
+    if (error.status) {
+      return getErrorMessageByStatus(error.status, message);
+    }
+  }
+  return message;
+}
+
+export default function ErrorMessage({ error }) {
+  const message = useMemo(() => getErrorMessage(error), [error]);
```

Comment:
Why bother with `useMemo` here? `getErrorMessage` doesn't feel that intensive.

Reply:
Fixed :+1: